content/doc/developer/testing/index.adoc

```diff
@@ -73,63 +88,12 @@ You can also apply the link:https://javadoc.jenkins.io/component/jenkins-test-ha
-    final String command = "echo hello";
-
-    // Create a new freestyle project with a unique name, with an "Execute shell" build step;
-    // if running on Windows, this will be an "Execute Windows batch command" build step
-    FreeStyleProject project = j.createFreeStyleProject();
-    Builder step = Functions.isWindows() ? new BatchFile(command) : new Shell(command); // <3>
-    project.getBuildersList().add(step);
-
-    // Enqueue a build of the project, wait for it to complete, and assert success
-    FreeStyleBuild build = j.buildAndAssertSuccess(project);
-
-    // Assert that the console log contains the output we expect
-    j.assertLogContains(command, build);
-  }
-
-  @Test public void pipelineEcho() throws Exception {
-    // Create a new Pipeline with the given (Scripted Pipeline) definition
-    WorkflowJob project = j.createProject(WorkflowJob.class); // <4>
-    project.setDefinition(new CpsFlowDefinition("node { echo 'hello' }", true)); // <5>
-
-    // Enqueue a build of the Pipeline, wait for it to complete, and assert success
-    WorkflowRun build = j.buildAndAssertSuccess(project);
-
-    // Assert that the console log contains the output we expect
-    j.assertLogContains("hello", build);
-  }
-}
-----
-<1> Declaring a `JenkinsRule` is the only requirement to automatically set up and tear down a Jenkins installation for each test method. You can disable this behavior for individual test methods by adding the link:https://javadoc.jenkins.io/component/jenkins-test-harness/?org/jvnet/hudson/test/WithoutJenkins.html[`@WithoutJenkins`] annotation.
-<2> link:https://javadoc.jenkins.io/component/jenkins-test-harness/?org/jvnet/hudson/test/BuildWatcher.html[`BuildWatcher`] captures the console log output for each build that runs in the test case, and writes it to standard output.
-<3> Try to ensure that your tests run on both Windows and Unix-like operating systems; the link:https://javadoc.jenkins.io/hudson/Functions.html#isWindows--[`isWindows()`] method can help here.
-<4> As the link:https://javadoc.jenkins.io/plugin/workflow-job/?org/jenkinsci/plugins/workflow/job/WorkflowJob.html[Pipeline project type] isn't included in Jenkins core, unlike Freestyle, we have to use the generic link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#createProject-java.lang.Class-[`createProject`] method with the `WorkflowJob` class, rather than a specific convenience method like link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#createFreeStyleProject[`createFreeStyleProject`].
-<5> The second parameter should *always* be set to `true`, as this enables plugin:script-security[script sandboxing].
+To see some working examples of plugin development,
+including use of `JenkinsRule` to test plugin code in the context of temporary Jenkins installations,
+try using one of the official [archetypes](https://github.com/jenkinsci/archetypes#usage).
```

Comment:
Detailed code examples here are hard to maintain and use, and there is no mechanical verification that they actually work. Archetypes are officially published, and the release/CI process ensures their tests compile and run. They are also more likely to be kept up to date with ongoing API improvements and best practices.